### ISSUE TITLE
Non-master release branch fails on the gh-pages step

### ISSUE THREAD

**Issue by andynygard** [CommBank/ci2#56]
There seems to be an issue where the build will try to push to `gh-pages` if we're building from a non-master release branch.

I think it's because the we use a pretty general script, `ci-push-branch.sh gh-pages`, which will push any release branch. However in the case of `gh-pages`, it only actually makes sense to do it from `master`.

See build: https://travis-ci.org/CommBank/uniform/builds/330680698#L2602

**Comment by jdcba** [comment]
Yeah this is an interaction between the project’s `.travis.yml` and CI2 when `RELEASE_BRANCHES` includes a non-master branch. CI2 appears to be doing what it’s told: pushing a release. So perhaps `.travis.yml` should only ask for a push when building master. However to make this generic, CI2 should probably include support for `Is_Master`.

https://github.com/CommBank/ci2/blob/75d3c9ca33caab6c0ccb30721261fa9bbbc01f84/src/main/bash/ci-push-branch.sh#L56

**Comment by andynygard** [comment]
Yeah, it's more of an edge-case in the way we interact with the API.

Mayhaps we look at introducing a `ci-push-gh-pages.sh` script, which wraps the above logic, to minimise the code in `.travis.yml` and keep things testable.
